app/src/main/java/nerd/tuxmobil/fahrplan/congress/favorites/StarredListFragment.java

```diff
@@ -301,12 +292,20 @@ private void askToDeleteAllFavorites() {
         }
     }
 
+    @SuppressWarnings("java:S5413")
     public void deleteAllFavorites() {
         MyApp.LogDebug(LOG_TAG, "deleteAllFavorites");
-        if (starredList == null) return;
+        if (starredList == null || starredList.isEmpty()) {
+            return;
+        }
+        appRepository.deleteAllHighlights();
         int count = starredList.size();
         for (int i = 0; i < count; i++) {
```

Comment:
Can you change this to a for each loop with a starredList.clear() after it?

Reply:
This does not work as far as I understand. `count` has to be constant because always the first element is removed.

Reply:
:heavy_check_mark: Rewritten as suggested in 8bee70b0aa1974e56610cde819f7df04ac8f92e7. Thanks for the improvement.